### ISSUE TITLE
Integrate the Test Explorer for running tests

### ISSUE THREAD

**Issue by 0xTim** [swift-server/vscode-swift#27]
We should use the test explorer to display tests available for running etc and make it easy to run all tests, a test class or individual test. In the end this should look something like:

![0a0ecb75da3eff362a7376d8bbd429a4f57232a0](https://user-images.githubusercontent.com/9938337/145438572-d88c5ad4-284e-4c4a-addd-5ab2feb80687.png)

This was done using https://github.com/MFranceschi6/vscode-swift-test-adapter so we should see if it makes sense to add that as a dependency or reimplement it depending on the experience it provides for the end user

**Comment by adam-fowler** [comment]
Doubt we can just merge this in, there is a good change it will clash with what is already there.

**Comment by adam-fowler** [comment]
Can we split this task into multiple PRs. I'd rather see it go in gradually than a ton of code just being dumped in the project. Not sure how it'll split up exactly. I guess we can start with 
- Generating the launch.json config. 
- Create interface class defining requirements for parsing of tests. Be it from `swift test --list-tests` or source kit-lsp.
- Link up with vscode test code

I'm kinda guessing at what these should be maybe once you've had a longer look at the test code you can come up with a better list @0xTim

**Comment by svanimpe** [comment]
I took a brief look at the existing extension. It works great, but it seems like it uses an older API. There's also quite a lot of output parsing involved in this 😕

**Comment by adam-fowler** [comment]
Instead of using `swift test --list-tests` which requires a recompile we can use the sourcekit-lsp server.
```swift
const tokenSource = new vscode.CancellationTokenSource();
try {
    const c = await this.languageClient!.sendRequest(
        langclient.DocumentSymbolRequest.type,
        this.languageClient!.code2ProtocolConverter.asDocumentSymbolParams(
            editor.document
        ),
        tokenSource.token
    );
    console.log(c);
} catch (e) {
    this.languageClient!.outputChannel.appendLine(
        `Could not fetch document: ${e}`
    );
    return [];
}
```

**Comment by adam-fowler** [comment]
The docs for the newer test framework are here https://vscode.trafficmanager.net/api/extension-guides/testing

**Comment by 0xTim** [comment]
That is a bizarre URL!

**Comment by adam-fowler** [comment]
> That is a bizarre URL!

Hmm not sure where that came from. Probably came from a search. Here is one from vscode https://code.visualstudio.com/api/extension-guides/testing

**Comment by adam-fowler** [comment]
Resolved with #154 and #156
